packages/core/src/constants.ts

```diff
@@ -63,11 +63,22 @@ export const CONTENTFUL_SECTION_ID = 'contentful-section'; // TODO: remove me on
 export const CONTENTFUL_CONTAINER_ID = 'contentful-container';
 export const CONTENTFUL_COLUMNS_ID = 'contentful-columns';
 export const CONTENTFUL_SINGLE_COLUMN_ID = 'contentful-single-column';
+export const CONTENTFUL_BUTTON_ID = 'button';
+export const CONTENTFUL_HEADING_ID = 'heading';
+export const CONTENTFUL_IMAGE_ID = 'image';
+export const CONTENTFUL_RICH_TEXT_ID = 'richText';
+export const CONTENTFUL_TEXT_ID = 'text';
 export const CONTENTFUL_SECTION_NAME = 'Section'; // TODO: remove me once all customers are using 2023-09-28 schema version
 export const CONTENTFUL_COLUMNS_NAME = 'Columns';
 export const CONTENTFUL_SINGLE_COLUMN_NAME = 'Column';
 export const CONTENTFUL_CONTAINER_NAME = 'Container';
+export const CONTENTFUL_BUTTON_NAME = 'Button';
```

Comment:
[nitpick]
Maybe, the time has come to rename or re-arrange those component-related constants as will expose all of this on the SDK and customers could use those.

**Idea 1: Create a map** (preferred)
```
export const CONTENTFUL_COMPONENT_IDS = {
    Button: 'button',
    // ...
    Container: 'contentful-container',
}
/** @deprecated will be replaced by CONTENTFUL_COMPONENT_IDS.Container in v5 */
export const CONTENTFUL_CONTAINER_ID = CONTENTFUL_COMPONENT_IDS.Container;

// --> Similar object literal for names
```

**Idea 2: Use a more specific suffix**
```
export const CONTENTFUL_COMPONENT_BUTTON_ID = 'button';
```

Reply:
I like the first idea.  I originally started making an Enum (similar to your map idea) but wasn't sure how to navigate the deprecation side and figured it would add more scope to the effort (such as variable replacement and UI changes) to switch all of the constants to use a new Enum instead.  I can do that as it would probably be better for the long term.

Reply:
I'm thinking about this more and I could see an object being better than an enum because we could organize related props together like this:
```ts
export const CONTENTFUL_COMPONENTS = {
  button: {
    id: 'button',
    name: 'Button',
    category: CONTENTFUL_DEFAULT_CATEGORY,
  },
  container: {
    id: 'contentful-container',
    name: 'Container',
    category: CONTENTFUL_COMPONENT_CATEGORY
  },
  // ...
}
```

Reply:
The constants only get exposed if they are reexported from the SDK package, but yes, putting some more organization around them by namespacing or enums is a good call.

Reply:
In the TS guild, we once agreed that we want to avoid enums if we don't need an exclusive feature from it:
https://contentful.atlassian.net/wiki/spaces/ENG/pages/4189094335/TypeScript+Standards#Enums